enhancements/network/primary-user-defined-networks-for-virtualization-workloads.md

```diff
@@ -0,0 +1,638 @@
+cluster in order to provide tenants with better isolation from each other,
+without the need for network policy.
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+OpenShift Virtualization is not supported on `MicroShift` clusters, hence,
+we do not plan to integrate with UDN on these type of clusters right now.
+
+It is tracked in this [jira issue](https://issues.redhat.com/browse/RFE-5289).
+
+Still, the passt integration is known for imposing some additional overhead
+which might not be suited for MicroShift. This could be mitigated by using
+bridge binding instead, which is more conservative in its resource consumption.
+
+OpenShift Virtualization is supported on single-node Openshift though; this
+integration should be fully supported in this deployment type.
+
+### Implementation Details/Notes/Constraints
+
+N/A
+
+### Risks and Mitigations
+
+#### Seamless live-migration risk
+
+Some packets will definitely be dropped during live-migration of the VM; while
+we want to have the migration as seamless as possible, just preserving the
+5 tuple of the existing connections is far from optimal.
+
+If the tests show that the traffic disruption takes too long, we should
+prioritize the
+[seamless live-migration epic](https://issues.redhat.com/browse/CNV-27147)
+which makes use of the multi requested-chassis OVN feature, in which a single
+logical switch port can be bound to multiple chassis (and have traffic
+forwarded to them).
+
+We have requested the core OVN team to ensure this feature is in fact usable on
+InterConnect clusters (which OpenShift is). This request is tracked in this FDP
+[issue](https://issues.redhat.com/browse/FDP-770).
+
+#### Allow user to configure the VMs interface desired IP address risk
+
+The proposed alternative for
+[allowing the user to specify the VM interface's IP](#allow-user-to-configure-the-vms-interface-desired-ip-address)
+requires a KubeVirt API update - and KubeVirt's feature freeze is just around
+the corner, on October 22nd.
+
+If we fail to get the code that updates the API by this date, we will have to
+pivot to the other alternative, even if it is less desired overall.
+
+### Drawbacks
+
+N/A
+
+## Test Plan
+
+- E2E upstream and downstream jobs covering supported features across multiple
+networks.
+- E2E tests ensuring VM live-migration preserving established TCP connections. 
+- E2E tests covering reachability to external networks (VM egress over UDN
+networks for Layer2 topologies). 
+- E2E tests covering a VM exposed as a `LoadBalancer` service is reachable from
+the outside world. 
+- Scale testing to determine limits and impact of multiple user-defined
+networks.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+There will be no dev or tech preview for this feature.
+
+### Tech Preview -> GA
+
+Targeting GA in OCP version `4.17.Z`.
```

Comment:
Do we need to elaborate why this is safe for users that expect only stabilization updates?

Reply:
Done.